internal/services/mysql/mysql_flexible_server_resource_test.go

```diff
@@ -611,6 +611,8 @@ resource "azurerm_private_dns_zone_virtual_network_link" "test" {
   private_dns_zone_name = azurerm_private_dns_zone.test.name
   virtual_network_id    = azurerm_virtual_network.test.id
   resource_group_name   = azurerm_resource_group.test.name
+
+  depends_on = [azurerm_subnet.test]
```

Comment:
how come this is needed?

Reply:
Recently I found testcases "TestAccDataSourceMySqlFlexibleServer_complete/TestAccMySqlFlexibleServer_complete/TestAccMySqlFlexibleServer_completeUpdate" are always failed in Teamcity Daily Run. After investigated, the root cause is that azurerm_private_dns_zone_virtual_network_link and azurerm_subnet cannot be deleted in the same time since azurerm_private_dns_zone_virtual_network_link is referring azurerm_subnet. So I added it to fix this issue.